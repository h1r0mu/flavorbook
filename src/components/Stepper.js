import React from "react";

import { makeStyles } from "@material-ui/core/styles";
import Stepper from "@material-ui/core/Stepper";
import Step from "@material-ui/core/Step";
import StepLabel from "@material-ui/core/StepLabel";
import PropTypes from "prop-types";
import { createMuiTheme } from "@material-ui/core/styles";

const useStyles = makeStyles((theme) => ({
  root: {
    marginTop: 8,
    width: "100%",
  },

  button: {
    marginTop: 50,
    marginLeft: 100,
    marginRight: 100,
  },

  instructions: {
    marginTop: 100,
    marginBottom: theme.spacing(1),
  },
}));

const HorizontalLinearStepper = (props) => {
  const classes = useStyles();

  const steps = [
    "Select Coffee Flavors ( Level1 )",
    "Select Coffee Flavors ( Level2 )",
    "Select Coffee Flavors ( Level3 )",
    "Register Coffee Flavors",
  ];
  return (
    <div className={classes.root}>
      <Stepper activeStep={props.level}>
        {steps.map((label) => (
          <Step key={label}>
            <StepLabel>{label}</StepLabel>
          </Step>
        ))}
      </Stepper>
      <div>{props.children}</div>
    </div>
  );
};

HorizontalLinearStepper.propTypes = {
  level: PropTypes.number.isRequired,
  children: PropTypes.element.isRequired,
};

export default HorizontalLinearStepper;
