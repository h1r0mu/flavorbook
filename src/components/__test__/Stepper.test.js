import React from "react";
import Stepper from "../Stepper.js";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const stepper = renderer
    .create(
      <Stepper level={0}>
        <div>dummy</div>
      </Stepper>
    )
    .toJSON();
  expect(stepper).toMatchSnapshot();
});
