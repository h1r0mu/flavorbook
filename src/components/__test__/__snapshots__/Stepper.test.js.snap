// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiPaper-root MuiStepper-root MuiStepper-horizontal MuiPaper-elevation0"
  >
    <div
      className="MuiStep-root MuiStep-horizontal"
    >
      <span
        className="MuiStepLabel-root MuiStepLabel-horizontal"
      >
        <span
          className="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              className="MuiStepIcon-text"
              textAnchor="middle"
              x="12"
              y="16"
            >
              1
            </text>
          </svg>
        </span>
        <span
          className="MuiStepLabel-labelContainer"
        >
          <span
            className="MuiTypography-root MuiStepLabel-label MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
          >
            Select Coffee Flavors ( Level1 )
          </span>
        </span>
      </span>
    </div>
    <div
      className="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
    >
      <span
        className="MuiStepConnector-line MuiStepConnector-lineHorizontal"
      />
    </div>
    <div
      className="MuiStep-root MuiStep-horizontal"
    >
      <span
        className="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
      >
        <span
          className="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiStepIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              className="MuiStepIcon-text"
              textAnchor="middle"
              x="12"
              y="16"
            >
              2
            </text>
          </svg>
        </span>
        <span
          className="MuiStepLabel-labelContainer"
        >
          <span
            className="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
          >
            Select Coffee Flavors ( Level2 )
          </span>
        </span>
      </span>
    </div>
    <div
      className="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
    >
      <span
        className="MuiStepConnector-line MuiStepConnector-lineHorizontal"
      />
    </div>
    <div
      className="MuiStep-root MuiStep-horizontal"
    >
      <span
        className="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
      >
        <span
          className="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiStepIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              className="MuiStepIcon-text"
              textAnchor="middle"
              x="12"
              y="16"
            >
              3
            </text>
          </svg>
        </span>
        <span
          className="MuiStepLabel-labelContainer"
        >
          <span
            className="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
          >
            Select Coffee Flavors ( Level3 )
          </span>
        </span>
      </span>
    </div>
    <div
      className="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
    >
      <span
        className="MuiStepConnector-line MuiStepConnector-lineHorizontal"
      />
    </div>
    <div
      className="MuiStep-root MuiStep-horizontal"
    >
      <span
        className="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
      >
        <span
          className="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiStepIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              className="MuiStepIcon-text"
              textAnchor="middle"
              x="12"
              y="16"
            >
              4
            </text>
          </svg>
        </span>
        <span
          className="MuiStepLabel-labelContainer"
        >
          <span
            className="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
          >
            Register Coffee Flavors
          </span>
        </span>
      </span>
    </div>
  </div>
  <div>
    <div>
      dummy
    </div>
  </div>
</div>
`;
